### PR TITLE
Keep right sidebar open, add Details action

### PR DESCRIPTION
**Pull request by PVince81**
- [x] TODO: unit tests

@jancborchardt please try it out

### CODE REVIEW

**Comment by jancborchardt** [comment]
Please also review @owncloud/designers

**Comment by PVince81** [comment]
Any update on this ? Would be good to getting the interaction sorted out.

**Comment by PVince81** [comment]
Rebased

**Comment by MorrisJobke** [comment]
```
13:52:00 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.FileList tests File actions redisplays actions when new actions have been registered FAILED
13:52:00 	Expected false to equal true.
13:52:00 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/apps/files/tests/js/filelistSpec.js:1993
```

**Comment by PVince81** [comment]
@MorrisJobke I haven't started with the unit tests. I will do so once I get confirmation that this is the way we want to go with the UI interaction.

**Comment by PVince81** [comment]
Any update ? Is this what we want ?
I'll finish this once a decision is made.

**Comment by MorrisJobke** [comment]
I would go that way.

Just talked with @tomneedham and it would be awesome to open the sidebar with the three dots click. Then we can but the actions in there. What about this? cc @owncloud/designers

**Comment by jancborchardt** [comment]
Why that? Then any other action requires a lot of mental work to find. Also the interface would switch around when you don’t really expect it.

(Btw we really need a short animation for the appearance of the sidebar.)

**Comment by MorrisJobke** [comment]
> Why that? Then any other action requires a lot of mental work to find. Also the interface would switch around when you don’t really expect it.

Because then it's also a two step thing to do everything. And nearly all actions (beside rename and download) will open the sidebar anyways.

**Comment by MorrisJobke** [comment]
This was just a thought.

**Comment by jancborchardt** [comment]
> nearly all actions (beside rename and download) will open the sidebar anyways.

See it the other way around: Actually _only_ the Versions action from the actions dropdown require the sidebar. The other 3 (Rename, Download and Delete) do not. ;)

**Comment by MorrisJobke** [comment]
> See it the other way around: Actually only the Versions action from the actions dropdown require the sidebar. The other 3 (Rename, Download and Delete) do not. ;)

I think both sides are valid and I'm fine with this. -> :+1: but a rebase is needed.

@jancborchardt do we want to do this? I think so.

**Comment by PVince81** [comment]
Rebased.

Should I go ahead and finish this then ?

**Comment by MorrisJobke** [comment]
> Should I go ahead and finish this then ?

@jancborchardt Please answer here

**Comment by PVince81** [comment]
Fixed this PR for the unit tests.

This is ready for review @jancborchardt @MorrisJobke @owncloud/designers @nickvergessen @schiesbn

**Comment by MorrisJobke** [comment]
Tested and works :+1:

@jancborchardt Can you provide an icon for this?

**Comment by jancborchardt** [comment]
I added an icon., and it works good. One remark though: Please put the Details action as the very first item in the actions list. The order should be like this:

* Details
* Download
* Rename
* Versions
* Delete

**Comment by PVince81** [comment]
Dammit... once again an order issue. This depends on the loading order of the apps.
I know you hate it that plugins can specify a sort order but I don't see how we can solve this without hard-coding id strings in core, which is even worse.

**Comment by PVince81** [comment]
General ordering problem discussed here: https://github.com/owncloud/core/issues/18280#issuecomment-137346384

**Comment by jancborchardt** [comment]
I think hard-coding the order in core is way better than again allowing apps to set their own arbitrary order parameter, which has been shown to fail with the app order.

**Comment by karlitschek** [comment]
Yes. We should add a detail item as discussed.

**Comment by oparoz** [comment]
Just add 10 to the order set by an app if you can make the difference between 3rd party and core apps and set the rest in stone.

**Comment by jancborchardt** [comment]
Well, delete should always be at the end. The other actions can be sorted alphabetically for example. I don’t see how introducing an app-defined order which devs need to set solves anything we can’t do here.

**Comment by oparoz** [comment]
@jancborchardt - Actually categories instead of specific order might be useful. Group file actions together, same for viewers/editors, etc.

**Comment by karlitschek** [comment]
ping. can be merge this?

**Comment by schiesbn** [comment]
Looks good to me. Only one small detail (but this could also be addressed in a separate bug fix). If the sidebar is open and I click on a empty space below the files list the preview icon and file name is gone but the "versions" and "sharing" tabs are still there which looks quite strange. Either we should close the side bar in this case or keep the last selected file selected until you chose a different file.

**Comment by PVince81** [comment]
Rebased.

I'll look into fixing the order stuff separately (tabs order and menu item order).

**Comment by karlitschek** [comment]
:+1:

**Comment by schiesbn** [comment]
:+1: beside the small things I mentioned here https://github.com/owncloud/core/pull/18273#issuecomment-140036095, maybe you can address them during the bug fix phase.

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -285,11 +290,25 @@
...
+			var self = this;
 			this.fileActions = fileActions;
 			if (!this.fileActions) {
 				this.fileActions = new OCA.Files.FileActions();
 				this.fileActions.registerDefaultActions();
 			}
+
+			if (this._detailsView) {
+				this.fileActions.registerAction({
+					name: 'Details',
```
Comment (nickvergessen) [inline apps/files/js/filelist.js]: not translated?

Comment (PVince81) [inline apps/files/js/filelist.js]: It should internally call t() with that. And now I realize that it cannot work.

I'll make a PR to add it as displayName (`name` is actually the action id, legacy stuff...)

Comment (PVince81) [inline apps/files/js/filelist.js]: @nickvergessen https://github.com/owncloud/core/pull/19266

**Pull request merged**